### ISSUE TITLE
About the data set.

### ISSUE THREAD

**Issue by mashumaro** [wrk226/pytorch-multimodal_sarcasm_detection#3]
It seems that little different from the original code by TF1, the origin code using the split train, valid and test set separately, but in this implement, the all set had been mixed and resplit. I try to use the three sets directly but it will make the model couldn't be trained. Sorry for the bother, but it really make me curious.

**Comment by wrk226** [comment]
I don't think using the original split will make the model unable to train. You need to check your code.

**Comment by yuchenWeng** [comment]
I found it also, and I changed to the original split, it run success, but F1 and acc decreased by 1.5% compare to original paper， is it some difference between tensorflow/pytorch

**Comment by wrk226** [comment]
> I found it also, and I changed to the original split, it run success, but F1 and acc decreased by 1.5% compare to original paper， is it some difference between tensorflow/pytorch

Yes, it because I didn't do exactly the same as the original code:
1. I initialized all weight by zero, instead of truncated normal distribution.
2. In the LSTM part, the author did the dropout on hidden state, but I did it on the cell itself. It because PyTorch didn't support that yet, you may check [here](https://github.com/pytorch/pytorch/issues/2166).

**Comment by mashumaro** [comment]
> > I found it also, and I changed to the original split, it run success, but F1 and acc decreased by 1.5% compare to original paper， is it some difference between tensorflow/pytorch
> 
> Yes, it because I didn't do exactly the same as the original code:
> 
> 1. I initialized all weight by zero, instead of truncated normal distribution.
> 2. In the LSTM part, the author did the dropout on hidden state, but I did it on the cell itself. It because PyTorch didn't support that yet, you may check [here](https://github.com/pytorch/pytorch/issues/2166).

Thanks for your kind answer, I fix my code and it worked. By the way, I would like to know which part has been initialized by zero.

**Comment by wrk226** [comment]
> > > I found it also, and I changed to the original split, it run success, but F1 and acc decreased by 1.5% compare to original paper， is it some difference between tensorflow/pytorch
> > 
> > 
> > Yes, it because I didn't do exactly the same as the original code:
> > 
> > 1. I initialized all weight by zero, instead of truncated normal distribution.
> > 2. In the LSTM part, the author did the dropout on hidden state, but I did it on the cell itself. It because PyTorch didn't support that yet, you may check [here](https://github.com/pytorch/pytorch/issues/2166).
> 
> Thanks for your kind answer, I fix my code and it worked. By the way, I would like to know which part has been initialized by zero.

I may be incorrect, they might not be initialized by zero, but what I want to say is that I didn't do any of the work on the initialize part(except the "early fusion" which described in the paper). So all of the layers are initialized by PyTorch's default setting, which may differ from the original code.
